packages/ragbits-document-search/src/ragbits/document_search/ingestion/providers/unstructured/utils.py

```diff
@@ -0,0 +1,124 @@
+import base64
+import io
+import os
+from typing import Optional
+
+from PIL import Image
+from unstructured.documents.elements import Element as UnstructuredElement
+
+from ragbits.core.llms.litellm import LiteLLM, LiteLLMOptions
+from ragbits.document_search.documents.document import DocumentMeta
+from ragbits.document_search.documents.element import TextElement
+
+
+def to_text_element(element: UnstructuredElement, document_meta: DocumentMeta) -> TextElement:
+    """
+    Converts unstructured element to ragbits text element
+
+    Args:
+        element: element from unstructured
+        document_meta: metadata of the document
+
+    Returns:
+        text element
+    """
+    return TextElement(
+        document_meta=document_meta,
+        content=element.text,
+    )
+
+
+def set_or_raise(name: str, value: Optional[str], env_var: str) -> str:
```

Comment:
The name, the description, and the actual behavior don't much, at least when it comes to my mental model.

- I would expect a function named `set_or_raise` to set a value and if it can't - to raise an exception. I also wouldn't know that it has anything to do with environmental variables
- I would expect a function described as "Checks if given environment variable is set and returns it or raises an error" to simply look for an env variable and if it's not present - to raise an exception
- In reality this is more like a fallback mechanism - first it tries to use the value given as an argument, if it's not present it tries to get it from the environment, and if it's not present there it throws an exception. The argument called "name" is not actually used anywhere other than the error message, and the error message makes it clear that the function is about checking arguments. I would have never gotten all this without reading the code. 

Maybe rename to something like `check_required_argument(value: Optional[str], arg_name: str, fallback_env: str)`

Reply:
This is actually a function that I simply moved from another class, it came without documentation so I wrote something myself, but I guess it wasn't enough. Changed to your suggestion.